odoo/tools/xml_utils.py

```diff
@@ -1,22 +1,38 @@
 # -*- coding: utf-8 -*-
+import base64
+from io import BytesIO
 from lxml import etree
-from odoo.tools.misc import file_open
+
 from odoo.exceptions import UserError
 
 def check_with_xsd(tree_or_str, stream):
     raise UserError("Method 'check_with_xsd' deprecated ")
 
+class odoo_resolver(etree.Resolver):
+    def __init__(self, env):
+        self.env = env
+
+    def resolve(self, url, id, context):
+        attachment = self.env['ir.attachment'].search([('name', '=', url)])
+        if attachment:
+            return self.resolve_string(base64.b64decode(attachment.datas), context)
+
 
-def _check_with_xsd(tree_or_str, stream):
+def _check_with_xsd(tree_or_str, stream, env=None):
+    parser = etree.XMLParser()
     if not isinstance(tree_or_str, etree._Element):
         tree_or_str = etree.fromstring(tree_or_str)
-    xml_schema_doc = etree.parse(stream)
+    # Enable searching in the ir.attachment's if env is given.
```

Comment:
"If self.env is given, the xsds can refer to other xsd by the name of their attachment, so we can also use something else for storing the xsd than the regular file system

Reply:
I find it less clear, less correct and too much verbose

Reply:
I added docstring, see if it is better